scripts/nbconverted/integrate-msigdb.py

```diff
@@ -0,0 +1,398 @@
+
+# coding: utf-8
+
+# # Create a hetnet of MSigDB genesets
+# 
+# This script was modified from https://github.com/dhimmel/integrate.
+# 
+# ## For the automatic interpretation of gene expression compression features
+# 
+# The script creates an MSigDB hetnet as described in the eLIFE publication _"Systematic integration of biomedical knowledge prioritizes drugs for repurposing"_ by [Himmelstein et al. 2017](https://doi.org/10.7554/eLife.26726)
+
+# In[1]:
+
+
+import os
+import csv
+import pandas as pd
+import seaborn as sns
+
+import hetio.hetnet
+import hetio.readwrite
+import hetio.stats
+
+
+# In[2]:
+
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# ## Define the metagraph and instantiate the graph
+
+# In[3]:
+
+
+kind_to_abbev = {
+    
+    # metanodes
+    'Gene': 'G',
+
+    'Cancer-Hallmarks': 'H',
+    'Positional-Gene-Sets': 'C1',
+    'Curated-Gene-Sets': 'C2',
+    'Motif-Gene-Sets': 'C3',
+    'Computational-Gene-Sets': 'C4',
+    'GO-Gene-Sets': 'C5',
+    'Oncogenic-Gene-Sets': 'C6',
+    'Immunologic-Gene-Sets': 'C7',
+    
+    # metaedges
+    'participates': 'p',
+}
+
+metaedge_tuples = [
+    ('Gene', 'Cancer-Hallmarks', 'participates', 'both'),
+    ('Gene', 'Positional-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Curated-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Motif-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Computational-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'GO-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Oncogenic-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Immunologic-Gene-Sets', 'participates', 'both'),
+]
+
+
+# In[4]:
+
+
+# Initialize the graph
+metagraph = hetio.hetnet.MetaGraph.from_edge_tuples(metaedge_tuples, kind_to_abbev)
+graph = hetio.hetnet.Graph(metagraph)
+
+
+# ## Get all genes found in MSigDB
+
+# In[5]:
+
+
+full_msigdb_file = os.path.join('data', 'full_msigdb_binary_matrix.tsv.bz2')
+msigdb_df = pd.read_table(full_msigdb_file, index_col=0)
+
+print(msigdb_df.shape)
+msigdb_df.head()
+
+
+# In[6]:
+
+
+# Ask the distribution of pathway sizes
+msigdb_df.sum(axis=1).sort_values().hist(bins=30);
+
+
+# In[7]:
+
+
+# What about the distribution of gene participation?
+msigdb_df.sum(axis=0).sort_values().hist(bins=30);
+
+
+# ## Gene Nodes
+
+# In[8]:
+
+
+# Load curated gene names from versioned resource 
+commit = 'a7362748a34211e5df6f2d185bb3246279760546'
+url = 'https://raw.githubusercontent.com/dhimmel/entrez-gene/{}/data/genes-human.tsv'.format(commit)
```

Comment:
I'd use [genes](https://github.com/cognoma/genes/blob/master/data/genes.tsv) from https://github.com/cognoma/genes and consider applying the [updater mapping](https://github.com/cognoma/genes/blob/master/data/updater.tsv). cognoma/genes is more up to date and the mapping is nice.

Reply:
I will use cognoma genes